### ISSUE TITLE
Version mismatch for the http and cached_network_image

### ISSUE THREAD

**Issue by gunhansancar** [renefloor/flutter_cached_network_image#81]
After `flutter upgrade` I couldn't run the project anymore because of the following reason.

[✓] Flutter (Channel dev, v0.9.6, on Mac OS X 10.13.6 17G65, locale en-GB)

```
The current Dart SDK version is 2.1.0-dev.6.0.flutter-8a919426f0.

Because no versions of cached_network_image match >0.4.2 and cached_network_image >=0.4.0-rc.1 <0.4.0 depends on flutter_cache_manager ^0.1.0-rc.1, cached_network_image >=0.4.0-rc.1 <0.4.0 or >0.4.2 requires flutter_cache_manager ^0.1.0-rc.1.
And because cached_network_image <0.4.0-rc.1 requires SDK version >=1.8.0 <2.0.0, cached_network_image <0.4.0-∞ or >0.4.2 requires flutter_cache_manager ^0.1.0-rc.1.
And because cached_network_image >=0.4.0 <0.4.1 depends on flutter_cache_manager ^0.1.0 and cached_network_image >=0.4.1 <0.4.2 depends on flutter_cache_manager ^0.1.1, cached_network_image <0.4.2-∞ or >0.4.2 requires flutter_cache_manager ^0.1.0-rc.1.
And because cached_network_image 0.4.2 depends on flutter_cache_manager ^0.1.2 which depends on http ^0.11.3+14, every version of cached_network_image requires http ^0.11.3+14 or flutter_cache_manager ^0.1.2.
Because flutter_cache_manager 0.1.2 depends on http ^0.11.3+14 and no versions of flutter_cache_manager match >0.1.2 <0.2.0, flutter_cache_manager ^0.1.2 requires http ^0.11.3+14.
Thus, every version of cached_network_image requires http ^0.11.3+14.
And because every version of flutter_test from sdk depends on http 0.12.0, flutter_test from sdk is incompatible with cached_network_image.
So, because edge_core depends on both cached_network_image any and flutter_test any from sdk, version solving failed.

pub get failed (1)
```

**Comment by gunhansancar** [comment]
For the time being I went back to Flutter v0.9.5 to be able to run the project.

Incase you're in the same position do the following 
1- Go to root of flutter sdk folder e.g. `cd development/flutter/`
2- Run the command `git checkout v"0.9.5"`

**Comment by hnvn** [comment]
To work around this issue:
```
dependency_overrides:
  http: ^0.12.0
```

**Comment by renefloor** [comment]
This issue only exists on the new master branch of flutter and not on the more stable beta branch. I just published new versions as flutter_cache_manager 0.2.0 and cached_network_image 0.5.0. You should use whatever version fits with your flutter SDK.
